client/tests/phpunit/Service/Audit/AwsAuditLogHandlerTest.php

```diff
@@ -0,0 +1,304 @@
+<?php declare(strict_types=1);
+
+namespace AppBundle\Service\Audit;
+
+use Aws\CloudWatchLogs\CloudWatchLogsClient;
+use Aws\Result;
+use Monolog\Handler\AbstractProcessingHandler;
+use Monolog\Logger;
+use PHPUnit\Framework\MockObject\MockObject;
+use PHPUnit\Framework\TestCase;
+
+class AwsAuditLogHandlerTest extends TestCase
+{
+    /** @var AwsAuditLogHandler */
+    private $sut;
+
+    /** @var MockObject | CloudWatchLogsClient */
+    private $cloudWatchClient;
+
+    const LOG_GROUP_NAME = 'audit-local';
+    const STREAM_NAME = 'DELETED_CLIENTS';
+
+    public function setUp(): void
+    {
+        $this->cloudWatchClient = $this
+            ->getMockBuilder(CloudWatchLogsClient::class)
+            ->disableOriginalConstructor()
+            ->addMethods(['putLogEvents', 'createLogStream', 'describeLogStreams'])
+            ->getMock();
+
+        $this->sut = new AwsAuditLogHandler($this->cloudWatchClient, self::LOG_GROUP_NAME);
+    }
+
+    /**
+     * @test
+     */
+    public function serviceIsInstanceOfAbstractHandler(): void
+    {
+        $this->assertInstanceOf(AbstractProcessingHandler::class, $this->sut);
+    }
+
+    /**
+     * @test
+     * @throws \Exception
+     */
+    public function ignoresRecordsWithoutEventName(): void
+    {
+        $record = [
+            'level' => Logger::NOTICE,
+            'message' => "Client Deleted",
+            'datetime' => new \DateTime('2018-09-02 13:42:23'),
+            'context' => ['type' => 'audit']
+        ];
+
+        $this
+            ->assertLogStreamWillNotBeCreated()
+            ->assertLogWillNotBePutOnAws();
```

Comment:
I say this every time but I really need to start writing custom assertions like this, it's a pleasure to read 👍

Reply:
It overkill on small tests, but wherever a lot of mock instantiating is required, describing it with a function seems the way to go 👍